### ISSUE TITLE
Detect settings automatically if profiles aren't matched

### ISSUE THREAD

**Issue by Lawrencemm** [conan-io/conan-clion-plugin#24]
From CLion's API it appears we can get the current toolchain and use that to pass settings to Conan command line if the user does not match profiles, of course retaining the ability to override with profiles.

I am currently looking into this.

**Comment by jgsogo** [comment]
Yes, this would be a nice-to-have enhancement, any advance on it? We can get help with the API documentation if needed.

**Comment by Lawrencemm** [comment]
I will have another look to refresh my memory, and get back to you.

I think it might require some communication with JetBrains to request eg.
They expose an enumeration for the compiler type, version etc. For each
configuration.

Maybe we can figure out what we need and make a case to them.

**Comment by jgsogo** [comment]
There is a slack channel to talk about CLion API, that should be the right channel to ask about this. Maybe you can get an invitation (ping @abusalimov)

**Comment by Lawrencemm** [comment]
Thanks! Will have a try.

**Comment by Lawrencemm** [comment]
@abusalimov Is there any way I could get an invite to the CLion API slack that I have heard about? I promise I will use the knowledge for good :slightly_smiling_face:

**Comment by abusalimov** [comment]
Yeah, sure. Please tell me your email so that I can send you an invitation.

-- 
Regards,
Eldar

**Comment by Lawrencemm** [comment]
lawrencem99@gmail.com

Thanks a lot!

**Comment by Lawrencemm** [comment]
@jgsogo This should probably be taken off milestone 1.2.0, it won't be ready.

**Comment by jgsogo** [comment]
Ok, I'll move it into the `next` milestone (meaning like it is _selected feature_ but we don't know when it is going to be available)... nevertheless, take it easy, one thing at a time. ☺️

**Comment by czoido** [comment]
Hi!

This issue targets the unmaintained 1.x plugin, which doesn't support Conan 2.x.

Please update to the current 2.x plugin from the [JetBrains Marketplace](https://plugins.jetbrains.com/plugin/11956-conan) and use Conan 2.x. If you run into any problems, feel free to open a new issue.

Closing as outdated. Thanks for the report!
